torch/_inductor/comms.py

```diff
@@ -183,18 +253,53 @@ def exposed_communication_time(collective_snode, remaining_snodes):
             if total_moves >= MOVE_LIMIT:
                 reorder_info.limiting_factor = "move limit"
                 continue
+
             for j in range(i - 1, -1, -1):
-                prev_snode = snodes[j]
+                prev_gsnode = gsnodes[j]
+                if len(prev_gsnode.snodes) == 0:
+                    continue
+
                 if j < max(0, i - PER_COLLECTIVE_PREFETCH_LIMIT):
                     reorder_info.limiting_factor = "prefetch limit"
                     break
-                if contains_collective(prev_snode):
+                if contains_collective(prev_gsnode):
                     reorder_info.limiting_factor = "collective ordering"
                     break
+
                 dep_names = OrderedSet([s.name for s in snode.unmet_dependencies])
-                if any(o.get_name() in dep_names for o in prev_snode.get_outputs()):
-                    reorder_info.limiting_factor = "data dependency"
-                    break
+                prev_outs = prev_gsnode.get_outputs()
+                data_dep = None
+                for o in prev_outs:
+                    if o.get_name() in dep_names:
+                        data_dep = o.get_name()
+                        break
+
+                if data_dep is not None:
+
+                    def is_groupable(prev_gsnode):
+                        # preserve ordering
+                        if contains_collective(prev_gsnode):
+                            return False
+
+                        if contains_ungroupable(prev_gsnode):
```

Comment:
nit: a bit arbitrary that "contains_collective" is a separate catetory than '"contains_ungroupable".  Would it be more readable to rename "contains_ungroupable" to "contains_gemm_like" ?